src/include/sof/ipc/common.h

```diff
@@ -13,14 +13,32 @@
 #include <sof/list.h>
 #include <sof/schedule/task.h>
 #include <sof/sof.h>
-#include <ipc/header.h>
 #include <user/trace.h>
 #include <stdbool.h>
 #include <stdint.h>
 
 struct dma_sg_elem_array;
 struct ipc_msg;
 
+/* generic IPC header regardless of ABI MAJOR type that is always 4 byte aligned */
+typedef uint32_t ipc_cmd_hdr;
```

Comment:
can we clarify what a 'header' is, there are parallel threads to remove this for Linux?

https://github.com/thesofproject/linux/pull/2856

@ujfalusi @RanderWang @ranj063 FYI

Reply:
@ujfalusi  please check history of header by git blame.  I think someone changed its usage.

Reply:
@plbossart IPC4 header is stored in IPC register IPCTDR while current IPC header is stored in memory windows.

Reply:
This is to represent an generic and opaque IPC message header irrelevant of the ABI major. It's intended to be passed around and then inspected by the IPC MAJOR layer only. i.e. it's completely opaque within FW and only used by IPC ABI major code.